Sources/Combine/Extensions/NotificationCenter/NotificationCenterPublisher+Unwrapping.swift

```diff
@@ -0,0 +1,27 @@
+//
+//  NotificationCenterPublisher+Unwrapping.swift
+//  Catalog
+//
+//  Created by Zvonimir Medak on 15.09.2022..
+//  Copyright © 2022 Infinum. All rights reserved.
+//
+
+import Foundation
+import Combine
+
+@available(iOS 13.0, *)
+public extension NotificationCenter.Publisher {
+
+    /// Unwraps the notifications for the specified type at the speficied key
+    ///
+    /// - Parameters:
+    ///  - key: Key at which the object will be unwrapped
+    ///  - type: Type of the object to which it will try to unwrap
+    ///
+    /// - Returns: AnyPublisher with the unwrapped value if there is one
+    func unwrapNotificationObject<T>(for key: String, ofType type: T.Type) -> AnyPublisher<T, Never> {
```

Comment:
```suggestion
    func unwrapNotificationObject<T>(for key: String, ofType type: T.Type) -> AnyPublisher<T, Never> {
```
I would usually return the exact type not the type-erased one.
If I remember correctly this is good article about the topic 😄 
https://swiftrocks.com/extending-creating-publishers-the-right-way

Reply:
This comment kind of applies to almost all of the functions here 😅